### DISCUSSION TITLE
Training xtts_v2, getting 5gb of model size vs 2gb of original one

### DISCUSSION THREAD

**Discussion by galazzo** [coqui-ai/TTS#3362] [General Q&A] [answered]
Hi,

training xtts_v2, following documentation guide then using gpt trainer I get a model having a final size more than twice the original one: 5GB vs 2GB.

Config parameters are different and if I try to use the original config the code is not able to manage it, than it means has been used a different trainer (I guess).

So I wonder I can I train xtts v2 using the same paramaters and tools used to produce the original one.

Thanks

**Comment by HobisPL** [comment]
If you can train the model in FP16, it will weigh 2GB, but FP16 might not work correctly because it doesn't display statistics. Unfortunately, I've already asked, and no one has responded to this question.
https://github.com/coqui-ai/TTS/discussions/3316

**Comment by erogol** [comment]
@Edresson any idea why?

**Comment by Edresson** [comment, marked as answer]
> Hi,
> 
> training xtts_v2, following documentation guide then using gpt trainer I get a model having a final size more than twice the original one: 5GB vs 2GB.
> 
> Config parameters are different and if I try to use the original config the code is not able to manage it, than it means has been used a different trainer (I guess).
> 
> So I wonder I can I train xtts v2 using the same paramaters and tools used to produce the original one.
> 
> Thanks

Hi @galazzo,

It is normal to get 5GB checkpoint during the training. The released checkpoint does not have the optimizer state but the saved during the training has it. If you like you can remove the optimizer states by loading the checkpoint and deleting the key "optimizer".

**Comment by shooding** [comment]
```
import torch
import os

# === Configuration ===
# Path to your full checkpoint (with optimizer state)
input_checkpoint_path = "run/training/GPT_XTTS_v2.0_LJSpeech_FT-March-21-2025_08+20PM-fb71e2a5/best_model.pth"

# Path to save the optimized, smaller checkpoint
output_model_path = "run/training/GPT_XTTS_v2.0_LJSpeech_FT-March-21-2025_08+20PM-fb71e2a5/optimized_model.pth"

# === Load checkpoint ===
print(f"Loading checkpoint: {input_checkpoint_path}")
checkpoint = torch.load(input_checkpoint_path, map_location="cpu")

# === Check contents ===
print(f"Checkpoint keys: {list(checkpoint.keys())}")

# === Remove optimizer state if present ===
if 'optimizer' in checkpoint:
    print("Removing optimizer state...")
    del checkpoint['optimizer']
else:
    print("No optimizer state found. Nothing to remove.")

# Save the checkpoint without deleted optimizer. 
# In xtts.py of coqui-ai-TTS the keys will auto ignored ["torch_mel_spectrogram_style_encoder", "torch_mel_spectrogram_dvae", "dvae"]
torch.save(checkpoint, output_model_path)
print(f"Saved shrinked model to: {output_model_path}")

# Optional: Show size reduction
orig_size = os.path.getsize(input_checkpoint_path) / (1024 * 1024)
new_size = os.path.getsize(output_model_path) / (1024 * 1024)
print(f"Original size: {orig_size:.2f} MB -> Optimized size: {new_size:.2f} MB")


```
